### DISCUSSION TITLE
Does anyone have a YACS config for the newer MaskRCNN-R50 baselines?

### DISCUSSION THREAD

**Discussion by alexander-soare** [facebookresearch/detectron2#3939] [Q&A]
I've spent the last hour or so trying to modify my code to support the new `LazyConfig` and I think I'm still going deeper into the forest. Has anyone got a YACS equivalent? I'm aware that it mightn't be possible to express everything in a LazyConfig in the older, less flexible YACS version but I just need it for fine-tuning so I doubt I need any of the extra custom funcionality.

Note: This is not a complaint about the new config. I think it's the right way to go, but unfortunately very bc breaking for anyone who has a lot of custom Detectron2 code.

**Comment by zhanghang1989** [comment]
YACS config may not have all the predefined settings. There may be some effort to add all the options for new training set up.

**Reply by alexander-soare** [reply]
Thanks for the response @zhanghang1989

> YACS config may not have all the predefined settings. There may be some effort to add all the options for new training set up.

Well the fine tuning training setup is whatever I define it to be right? I don't have to follow the training scheme used to train the model from scratch. So if I already have a setup that uses the original Detectron2 weights for MaskRCNN , I should be able to just swap out the weights.

Indeed I tried this by taking `mask_rcnn_R_50_fpn.yaml` and setting `MODEL.WEIGHTS` to point to the new weights, but I found that there was something wrong with the fine tuning loss curve (purple is old weights, yellow is new):

![image](https://user-images.githubusercontent.com/16543381/152762234-51108add-27f4-4b12-bb9e-5e065122bd53.png)

so I'm guessing there's something different about the model/input configuration, but since the configs look so different I couldn't figure out what.

**Reply by zhanghang1989** [reply]
the model architecture is not exactly the same. The box and mask heads are heavier than before, so the weights can't be loaded correctly without updating the config

**Reply by alexander-soare** [reply]
Great, thanks. Might try to just modify the YACS for those.
